util/exercise/src/cmd/configure.rs

```diff
@@ -29,22 +29,28 @@ fn get_user_config(exercise_name: &str, config_content: &Value) -> Result<Value>
         Uuid::new_v4().to_hyphenated().to_string()
     };
 
-    let core = false;
-
-    let unlocked_by = loop {
+    let unlocked_by: Option<String> = loop {
         let default_value = if let Some(existing_config) = existing_config {
-            get!(existing_config, "unlocked_by", as_str)
+            get!(existing_config, "unlocked_by")
+                .as_str()
+                .map(|s| s.to_string())
         } else {
-            "hello-world"
+            Some("hello-world".to_string())
```

Comment:
Simply removing the two `to_string()` would not work - the `unlocked_by` variable expects `Option<String>`. If I change it to the `Option<&str>` (along with some additional changes), I am blocked by the line 62 `break Some(&user_input);`, since `user_input` does not live long enough to make a borrow (does not leave outside the loop). 
Maybe you have some sort of suggestion for the problem? I have met a lot of moments, where I need an additional allocation, just so that the returned value is `String` and not `&str`, and I feel like this approach is not very idiomatic (given how many crates claim to be zero-copy)

Reply:
https://github.com/ZapAnton/rust/pull/1: if you merge it, it will update this PR